packages/vite/src/node/server/middlewares/static.ts

```diff
@@ -93,3 +108,29 @@ export function serveRawFsMiddleware(): Connect.NextHandleFunction {
     }
   }
 }
+
+function renderFsRestrictedHTML(root: string) {
+  // to have syntax highlighting and autocompletion in IDE
+  const html = String.raw
+  return html`
+    <body>
+      <h1>403 Restricted</h1>
+      <p>
+        For security concern, accessing files outside of workspace root
+        (<code>${root}</code>) is restricted since Vite v2.3.x
+      </p>
+      <p>
+        Refer to docs
+        <a href="https://vitejs.dev/config/#server-fsserveroot"
+          >https://vitejs.dev/config/#server-fsserveroot</a
+        >
```

Comment:
Personally I like more to do this:

```suggestion
        <a href="https://vitejs.dev/config/#server-fsserveroot">
          https://vitejs.dev/config/#server-fsserveroot
        </a>
```

Reply:
It wasn't me, it's prettier 👀

Reply:
Yeah, but prettier allow both as far as I know  
If you manually rewrite it, then prettier will not convert it back, that's the trick 🙂

Reply:
🎉